server/models.py

```diff
@@ -0,0 +1,581 @@
+            CREATE OR REPLACE FUNCTION human_sort(text)
+            RETURNS text[] AS
+            $BODY$
+            /* Split the input text into contiguous chunks where no numbers appear,
+                and contiguous chunks of only numbers. For the numbers, add leading
+                zeros to 20 digits, so we can use one text array, but sort the
+                numbers as if they were big integers.
+                For example, human_sort('Run 12 Miles') gives
+                        {'Run ', '00000000000000000012', ' Miles'}
+            */
+            select array_agg(
+                case
+                when a.match_array[1]::text is not null
+                    then a.match_array[1]::text
+                else lpad(a.match_array[2]::text, 20::int, '0'::text)::text
+                end::text)
+                from (
+                select regexp_matches(
+                    case when $1 = '' then null else $1 end, E'(\\\\D+)|(\\\\d+)', 'g'
+                ) AS match_array
+                ) AS a
+            $BODY$
+            LANGUAGE sql IMMUTABLE;
+            COMMIT;
+        """
+    ),
+)
+
+
+class UTCDateTime(TypeDecorator):  # pylint: disable=abstract-method
+    # Store with no timezone
+    impl = DateTime
+
+    # Ensure UTC timezone on write
+    def process_bind_param(self, value, dialect):
+        if value:
+            assert (
+                value.tzinfo == timezone.utc
+            ), "All datetimes must have UTC timezone - use datetime.now(timezone.utc)"
+        return value
+
+    # Repopulate UTC timezone on read
+    def process_result_value(self, value, dialect):
+        return value and value.replace(tzinfo=timezone.utc)
+
+
+class BaseModel(Base):
+    __abstract__ = True
+    created_at = Column(
+        UTCDateTime, default=lambda: dt.now(timezone.utc), nullable=False
+    )
+    updated_at = Column(
+        UTCDateTime,
+        default=lambda: dt.now(timezone.utc),
+        onupdate=lambda: dt.now(timezone.utc),
+        nullable=False,
+    )
+
+
+# on-delete-cascade is done in SQLAlchemy like this:
+# https://stackoverflow.com/questions/5033547/sqlalchemy-cascade-delete
+
+
+class Organization(BaseModel):
+    id = Column(String(200), primary_key=True)
+    name = Column(String(200), nullable=False, unique=True)
+
+    elections = relationship(
+        "Election",
+        back_populates="organization",
+        passive_deletes=True,
+        order_by="Election.election_name",
+    )
+
+
+# Election is a slight misnomer - this model represents an Election.
+class Election(BaseModel):
+    id = Column(String(200), primary_key=True)
+
+    election_name = Column(String(200), nullable=False)
+    election_date = Column(String(40), nullable=False)
+    polls_open = Column(String(5), nullable=False)
+    polls_close = Column(String(5), nullable=False)
+    polls_timezone = Column(String(4), nullable=False)
+    certification_date = Column(String(40), nullable=False)
```

Comment:
these dates should be proper datetime objects. Better convention is

`polls_open_at` --> which by name indicates it's a timestamp. See the `updated_at` field in the base class.

Dates should be stored purely as UTC, no timezone. Can cast to the viewer's timezone. I can see possibly a purpose to store the timezone where the election is happening, but the timestamps should all be UTC.